### PR TITLE
[C-1714] Improve mobile play count

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

- Consolidates playCount logic by removing custom mobile playCount to use common/web saga
- Removes constraint for preventing track owner to increase play count
- Fixes cases where tracking was over counted due to changes with the track object

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-1714-fix-mobile-play-count

File: packages/mobile/src/components/audio/Audio.tsx [hunk 1]
```diff
@@ -267,21 +268,16 @@ export const Audio = () => {
     const duration = await TrackPlayer.getDuration()
     const position = await TrackPlayer.getPosition()
 
-    // Replicates logic in dapp.
-    // TODO: REMOVE THIS ONCE BACKEND SUPPORTS THIS FEATURE
+    // Debounce logging a listen, update the state variable appropriately onSuccess and onFailure
```
Comment (sliptype) [inline packages/mobile/src/components/audio/Audio.tsx:271]: Is this comment still accurate?

**Review by sliptype — APPROVED** [review]
Good stuff! Let's keep an eye on amplitude when this goes out, I'm interested how mobile play counts will be affected

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-1714-fix-mobile-play-count

File: packages/mobile/src/components/audio/Audio.tsx (hunk 1 above)
Comment (dylanjeffers) [reply packages/mobile/src/components/audio/Audio.tsx:271]: yeah good call, not really

**Pull request merged**